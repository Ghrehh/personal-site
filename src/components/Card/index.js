// @flow
import * as React from 'react';
import combineClass from 'utils/combine-class';
import styles from './styles.module.css';

type Props = {
  children: React.Node,
  className?: string
};

const Card = (props: Props) => (
  <section className={styles.card}>
    <div className={combineClass(styles.cardInner, props.className)}>
      {props.children}
    </div>
  </section>
);

export default Card;
