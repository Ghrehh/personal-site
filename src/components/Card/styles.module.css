.card {
  max-width: 580px;
  padding: 40px;
  border: 1px solid var(--grey);
  border-radius: 3px;
  background-color: white;
  text-align: center;
  margin: 80px auto 50px;
  box-sizing: border-box
}

@media (max-width: 620px) {
  .card {
    margin: 40px 20px 20px;
  }
}
