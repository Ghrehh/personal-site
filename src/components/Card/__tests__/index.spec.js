import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Card from '../';

describe('Card', () => {
  let testContext = {};
  let component = (
    <Card>
      <p>Foo</p>
      <p>bar</p>
    </Card>
  );

  beforeEach(() => {
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(component);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
