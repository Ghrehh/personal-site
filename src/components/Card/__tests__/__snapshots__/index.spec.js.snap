// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders the component correctly 1`] = `
<section
  className="card"
>
  <div
    className="cardInner foo"
  >
    <p>
      Foo
    </p>
    <p>
      bar
    </p>
  </div>
</section>
`;
